src/components/layout/side-menu/SideMenu.js

```diff
@@ -0,0 +1,57 @@
+import { AppBar } from '@material-ui/core';
+import {  
+  Home,  
+  Devices,
+  Sensors,
+  History,
+  LocalHospital,
+} from '@mui/icons-material';
+import { ToggleButton } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { StyledToggleButtonGroup, useStyles } from './SideMenu.styles';
+
+export default function SideMenu() {
+  const [selected, setSelected] = useState('/');
+  const styles = useStyles();
+  const history = useHistory();
+  const location = useLocation();
+
+  const handleSelected = (event, newSelected) => {    
+    if (newSelected === null) {
+      return;
+    }
+    setSelected(newSelected);    
+    history.push(`/sistema${newSelected}`);    
+  };
+  
```

Comment:
```suggestion
  useEffect(() => {
    setSelected(location.pathname.replace('/sistema', ''));
  }, [location.pathname]);
```
Se você adicionar essas linhas, o estado do menu lateral vai acompanhar a url também, logo o estado do menu lateral vai se manter correto mesmo se você recarregar a página.